Client/mods/deathmatch/logic/CSingularFileDownload.cpp

```diff
@@ -65,6 +66,15 @@ void CSingularFileDownload::CallFinished(bool bSuccess)
         CLuaArguments Arguments;
         Arguments.PushString(GetShortName());            // file name
         Arguments.PushBoolean(bSuccess);                 // Completed successfully?
+        if (m_pRequestResource)
+        {
+            Arguments.PushResource(m_pRequestResource);  // Resource that called downloadFile
+        }
+        else
+        {
+            Arguments.PushBoolean(false);                 // or false
+        }

```

Comment:
@TheNormalnij, when is `m_pRequestResource` null?

Reply:
I think that m_pRequestResource can be deleted during downloading